src/shared/components/product-card/vertical-card/ProductVerticalCard.jsx

```diff
@@ -1,8 +1,61 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useAlert, useAuth, useWishList } from '../../../context';
+import {
+  deleteProductFromWishList,
+  saveProductInWishList,
+} from '../../../services/product.service';
 import './ProductVerticalCard.scss';
+import { useNavigate } from 'react-router-dom';
+import {
+  INCREMENT_WISH_COUNT,
+  DECREMENT_WISH_COUNT,
+} from '../../../actions/types';
+
+export const ProductVerticalCard = ({
+  product,
+  actionBtnText,
+  isInWishList,
+  removeProductFromWishList,
+}) => {
+  const { _id, store, price, imgUrl, categoryName, rating } = product;
+  const { state: authState } = useAuth();
+  const { wishListDispatch } = useWishList();
+  const { setAlertMessage } = useAlert();
+
+  const [wishListToggle, setWishListToggle] = useState(isInWishList);
+
+  useEffect(() => {
+    setWishListToggle(isInWishList);
+  }, [isInWishList]);
+
+  const navigate = useNavigate();
+
+  const addRemoveWishList = async () => {
+    setWishListToggle(!wishListToggle);
+    try {
+      if (authState.isUserLoggedIn) {
+        const data = !wishListToggle
+          ? await saveProductInWishList(product)
+          : await deleteProductFromWishList(_id);
+
+        if (data.status === 201 && !wishListToggle) {
+          wishListDispatch({ type: INCREMENT_WISH_COUNT });
+          setAlertMessage('Added to wishlist');
+        }
+
+        if (data.status === 200 && wishListToggle) {
+          wishListDispatch({ type: DECREMENT_WISH_COUNT });
```

Comment:
I think, you can send the wislistDispatch type just as a string also, why need to create separate file for it and then importing it from there..

Reply:
I have created type.js file 
1. Here we can check all available action types in app
2. If we are  dispatching same action in multiple places , we can easily import it anywhere  and it will avoid spelling mistakes in action type name